nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -278,6 +289,9 @@ public String getName() {
     }
 
     public void set(String key, Object value) {
+        if (value instanceof Calendar || value instanceof Date) {
```

Comment:
Actually the Nuxeo Java Client accepts all types of object to be set in document property. So it can be primitive types e.g. int, long, float, but it can also be any POJO. Therefore, I try to limit the verification to class types related to date here.